src/plugin.js

```diff
@@ -0,0 +1,202 @@
+            url = urlToRequest(url);
+
+            const importUrlPrefix = getImportPrefix(
+              options.loaderContext,
+              options.importLoaders
+            );
+
+            runtimeCode = `exports.i(require(${stringifyRequest(
+              options.loaderContext,
+              importUrlPrefix + url
+            )}), ${JSON.stringify(mediaQuery)});\n`;
+          } else {
+            runtimeCode = `exports.push([module.id, ${JSON.stringify(
+              `@import url(${url});`
+            )}, ${JSON.stringify(mediaQuery)}]);`;
+          }
+
+          if (!alreadyImported[url]) {
+            result.messages.push({
+              pluginName,
+              type: 'modify-module',
+              modifyModule: (moduleObj) => {
+                // eslint-disable-next-line no-param-reassign
+                moduleObj.runtime = `${moduleObj.runtime}${runtimeCode}\n`;
+
+                return moduleObj;
+              },
+            });
+
+            alreadyImported[url] = true;
+          }
+
+          rule.remove();
+        });
+      }
+
+      if (options.url) {
+        let index = 0;
+
+        css.walkDecls((decl) => {
+          if (!/url\(/i.test(decl.value)) {
+            return decl;
+          }
+
+          const parsedValue = valueParser(decl.value);
+
+          // eslint-disable-next-line no-param-reassign
+          decl.value = parsedValue
+            .walk((node) => {
+              if (
+                node.type !== 'function' ||
+                node.value.toLowerCase() !== 'url' ||
+                node.nodes.length === 0
+              ) {
+                return;
+              }
+
+              const [urlNode] = node.nodes;
+              const url = urlNode.value.trim().replace(/\\[\r\n]/g, '');
+
+              // Skip empty URLs
+              // Empty URL function equals request to current stylesheet where it is declared
+              if (url.length === 0) {
+                return;
+              }
+
+              if (!isUrlRequest(url)) {
+                return;
+              }
+
+              // Remove spaces before and after
+              // eslint-disable-next-line no-param-reassign
+              node.before = '';
+              // eslint-disable-next-line no-param-reassign
+              node.after = '';
+
+              const splittedURL = url.split(/(\?)?#/);
+              const [normalizedURL] = splittedURL;
+
+              const requestedURL = urlToRequest(normalizedURL);
+              const placeholder = `___CSS_LOADER_IMPORT_URL_PLACEHOLDER___${index}___`;
+
+              urlNode.value = placeholder;
+              // Strip quotes, they will be re-added if the module needs them
+              urlNode.quote = '';
+
+              let hasURLEscapeRuntimeCode = false;
+
+              result.messages.push({
+                pluginName,
+                type: 'modify-module',
+                modifyModule: (moduleObj, loaderContext) => {
```

Comment:
```js
{
   plugin,
   type: 'module',
   module (result, loader) {}
}
```

Reply:
@michael-ciniawsky `module` is misleading, I spent a lot of time in `stylelint` and work with messages, better use name which reflects the action not essence. Because we can add in future example `modify-module-urls`, `module-module-imports` and etc. It's like naming methods in classes.

Reply:
```js
{
  type: 'module'
  ...,
  modify || modifyModule () {}
}
```

or did you mean the `type`? The 'problem' with mutiple type names is that it just adds additional stuff to learn and checks witin the loader. `module` is generic compared to `modify-module` which to some point implies there might be a `*-module` type, but agreed it is a bad `{Function}`  name.

Reply:
@michael-ciniawsky why `module` better `modify-module`? I see only misleading name for `module`.